rocketpy/Function.py

```diff
@@ -2024,9 +2024,267 @@ def integral(self, a, b, numerical=False):
+
+        Returns
+        -------
+        result : Function
+            A Function object which gives the derivative of self.
+        """
+        # Check if Function object source is array
+        if isinstance(self.source, np.ndarray):
+            # Operate on grid values
+            Ys = np.diff(self.source[:, 1]) / np.diff(self.source[:, 0])
+            Xs = self.source[:-1, 0] + np.diff(self.source[:, 0]) / 2
+            source = np.concatenate(([Xs], [Ys])).transpose()
+
+            # Retrieve inputs, outputs and interpolation
+            inputs = self.__inputs__[:]
+            outputs = "d(" + self.__outputs__[0] + ")/d(" + inputs[0] + ")"
+            outputs = "(" + outputs + ")"
+            interpolation = "linear"
+
+            # Create new Function object
+            return Function(source, inputs, outputs, interpolation)
+        else:
+            return Function(lambda x: self.differentiate(x))
+
+    def integralFunction(self, lower=None, upper=None, datapoints=100):
+        """Returns a Function object representing the integral of the Function object.
+
+        Parameters
+        ----------
+        lower : scalar, optional
+            The lower limit of the interval in which the function is to be
+            plotted. If the Function is given by a dataset, the default
+            value is the start of the dataset.
+        upper : scalar, optional
+            The upper limit of the interval in which the function is to be
+            plotted. If the Function is given by a dataset, the default
+            value is the end of the dataset.
+        datapoints : int, optional
+            The number of points in which the integral will be evaluated for
+            plotting it, which draws lines between each evaluated point.
+            The default value is 100.
+
+        Returns
+        -------
+        result : Function
+            The integral of the Function object.
+        """
+        # Check if lower and upper are given
+        if lower is None:
+            if isinstance(self.source, np.ndarray):
+                lower = self.source[0, 0]
+            else:
+                raise ValueError("Lower limit must be given if source is a function.")
+        if upper is None:
+            if isinstance(self.source, np.ndarray):
+                upper = self.source[-1, 0]
+            else:
+                raise ValueError("Upper limit must be given if source is a function.")
+
+        # Create a new Function object
+        xData = np.linspace(lower, upper, datapoints)
+        yData = np.zeros(datapoints)
+        for i in range(datapoints):
+            yData[i] = self.integral(lower, xData[i])
+        return Function(np.concatenate(([xData], [yData])).transpose(), 
+                    inputs=self.__inputs__, 
+                    outputs=[o + " Integral" for o in self.__outputs__])
+
+    def invert(self, lower=None, upper=None, datapoints=100):
```

Comment:
I believe this docstring needs to be a bit clearer in terms of what happens with callable defined Functions. From what I understand, it sets it as a discretely defined Function before inverting. What if we wrapped `findOptimalInput` in this case, instead?

I know this is much more costly, but it is more rigorous. If the user wants a faster, less accurate version, he/she can use `Function.setDiscrete` before calling `Function.invert`. 

@gautamsaiy and @phmbressan, what do you think?

Reply:
I think this a great idea. I will make sure to add it.